tools/socktap/main.cpp

```diff
@@ -86,10 +90,14 @@ int main(int argc, const char** argv)
             return 1;
         }
 
-        auto signal_handler = [&io_service](const boost::system::error_code& ec, int signal_number) {
+        auto signal_handler = [&io_service, &link_layer](const boost::system::error_code& ec, int signal_number) {
             if (!ec) {
                 std::cout << "Termination requested." << std::endl;
                 io_service.stop();
+#ifdef SOCKTAP_WITH_AUTOTALKS
+                if (nullptr != dynamic_cast<AutotalksLink*>(link_layer.get()))      // Deinitialize Autotalks device only if its link layer used
+                    autotalks_device_deinit();
```

Comment:
Move `autotalks_device_deinit` to the (yet missing) destructor of `AutotalksLink`. Then no changes to *main.cpp* are necessary at all.